popup.js

```diff
@@ -9,42 +9,47 @@ function getSearchResultsElement() {
 
 document.addEventListener('DOMContentLoaded', function() {
   focusInput();
+  const browser = chrome || browser;
   const searchInput = getSearchInputElement();
   searchInput.addEventListener('keyup', async function(event) {
     const query = searchInput.value;
-    const searchResults = await searchBookmarks(query);
+    const bookmarkSearchResults = await searchBookmarks(query);
     // refine results
-    const refinedResults = refineResults(searchResults, query);
+    const refinedResults = refineResults(bookmarkSearchResults, query);
     updateSearchText(refinedResults);
     updateLinkEventListeners();
     if (event.key === "Enter") {
       // go to first event in the list
       const top_result = refinedResults[0];
-      window.open(top_result.url)
+      browser.tabs.create({url: top_result.url, active: true});
+      window.close();
     }
   });
 }, false);
 
 
-function refineResults(searchResults, query) {
-  return searchResults.filter(result => {
+function refineResults(bookmarkSearchResults, query) {
+  return bookmarkSearchResults.filter(result => {
     // cuts bookmark Title down to a substring for closer matching
     const queryLen = query.length;
     const queryLower = query.toLowerCase();
     const bookmarkTitle = result.title.substring(0, queryLen).toLowerCase();
-    return bookmarkTitle.includes(queryLower) & result.hasOwnProperty('url');
+    return bookmarkTitle.includes(queryLower) && (typeof result.url != 'undefined');
```

Comment:
is `(typeof result.url != 'undefined')` necessary over just `result.url`. js does coalescing to boolean

Reply:
@wojm that suggestion ends up not working because FF bookmark objects have the `url` property. Looks like `{title: 'some bookmark', url: undefined}`, so it's not enough to just check if it exists. This was settled on after testing in the browser.